btrack/utils.py

```diff
@@ -221,6 +226,88 @@ def nans_idx(x):
     return data, properties, graph
 
 
+def napari_to_tracks(
+    data: npt.NDArray,
+    properties: Optional[dict[str, npt.ArrayLike]],
+    graph: Optional[dict[int, list[int]]],
+) -> list[btypes.Tracklet]:
+    """Convert napari Tracks to a list of Tracklets.
+
+    Parameters
+    ----------
+    data : array (N, D+1)
+        Coordinates for N points in D+1 dimensions. ID,T,(Z),Y,X. The first
+        axis is the integer ID of the track. D is either 3 or 4 for planar
+        or volumetric timeseries respectively.
+    properties : dict {str: array (N,)}
+        Properties for each point. Each property should be an array of length N,
+        where N is the number of points.
+    graph : dict {int: list}
+        Graph representing associations between tracks. Dictionary defines the
+        mapping between a track ID and the parents of the track. This can be
+        one (the track has one parent, and the parent has >=1 child) in the
+        case of track splitting, or more than one (the track has multiple
+        parents, but only one child) in the case of track merging.
+
+    Returns
+    -------
+    tracks : list[btypes.Tracklet]
+        A list of tracklet objects from BayesianTracker.
+
+    """
+
+    if data.shape[1] == Dimensionality.FIVE:
+        track_id, t, z, y, x = data.T
+    elif data.shape[1] == Dimensionality.FOUR:
+        track_id, t, y, x = data.T
+        z = np.zeros_like(x)
+    else:
+        raise ValueError(
+            "Data must have either 4 (ID, t, y, x) or 5 (ID, t, z, y, x) columns, "
+            f"not {data.shape[1]}"
+        )
+
+    # Create all PyTrackObjects
+    objects_dict = {
+        "ID": properties.get("refs", np.arange(track_id.size)),
```

Comment:
I'm not sure about using `refs` here. These are really pointers into the original set of objects. But since that was lost on converting to a napari layer, I'm not sure it makes sense to reuse them here, since they may not point to the correct objects in the list anymore.

Reply:
I'm on the fence about this. Perhaps we leave it in for now. My assumption is that `refs == p.arange(track_id.size))` in most cases

Reply:
> These are really pointers into the original set of objects

oh I see, in that case I think I'd also lean toward creating new ids when converting from napari, but happy to leave it in for now if you prefer

Reply:
I would just use the new ids. I think having the refs complicates things